src/ethereum/vm/instructions.py

```diff
@@ -44,21 +68,313 @@ def add(evm: Evm) -> None:
     """
     evm.gas_left = subtract_gas(evm.gas_left, GAS_VERY_LOW)
 
-    x = pop(evm.stack)
-    y = pop(evm.stack)
-
+    x, y = pop_multiple(evm.stack, 2)
     val = x.wrapping_add(y)
 
     push(evm.stack, val)
 
 
+def sub(evm: Evm) -> None:
+    """
+    Subtracts the top two elements of the stack, and pushes the result back
+    on the stack.
+
+    Parameters
+    ----------
+    evm : `Evm`
+        The current EVM frame.
+
+    Raises
+    ------
+    StackUnderflowError
+        If `len(stack)` is less than `2`.
+    OutOfGasError
+        If `evm.gas_left` is less than `GAS_VERY_LOW`.
+    """
+    evm.gas_left = subtract_gas(evm.gas_left, GAS_VERY_LOW)
+
+    x, y = pop_multiple(evm.stack, 2)
+    val = x.wrapping_sub(y)
+
+    push(evm.stack, val)
+
+
+def mul(evm: Evm) -> None:
+    """
+    Multiply the top two elements of the stack, and pushes the result back
+    on the stack.
+
+    Parameters
+    ----------
+    evm : `Evm`
+        The current EVM frame.
+
+    Raises
+    ------
+    StackUnderflowError
+        If `len(stack)` is less than `2`.
+    OutOfGasError
+        If `evm.gas_left` is less than `GAS_LOW`.
+    """
+    evm.gas_left = subtract_gas(evm.gas_left, GAS_LOW)
+
+    x, y = pop_multiple(evm.stack, 2)
+    val = x.wrapping_mul(y)
+
+    push(evm.stack, val)
+
+
+def div(evm: Evm) -> None:
+    """
+    Integer division of the top two elements of the stack. Pushes the result
+    back on the stack.
+
+    Parameters
+    ----------
+    evm : `Evm`
+        The current EVM frame.
+
+    Raises
+    ------
+    StackUnderflowError
+        If `len(stack)` is less than `2`.
+    OutOfGasError
+        If `evm.gas_left` is less than `GAS_LOW`.
+    """
+    evm.gas_left = subtract_gas(evm.gas_left, GAS_LOW)
+
+    x, y = pop_multiple(evm.stack, 2)
+    res = (x // y) if y != 0 else U256(0)
```

Comment:
```suggestion
    if y == 0:
      push(evm.stack, U256(0))
      return
    
    result = x // y
    
    push(evm.stack, result)
```

Reply:
I'd rather be *super* clear about things. Writing it like this emphasizes that the EVM does something weird and returns `0` when dividing by 0.